### ISSUE TITLE
Meaning of change-file-permissions-at?

### ISSUE THREAD

**Issue by bjorn3** [bytecodealliance/preview2-prototyping#37]
What is the meaning of change-file-permissions-at? Does it only affect the permissions for the owner, or for everyone? If the later could you add a method to at least allow revoking read permission for everyone other than the file owner? That is pretty important for sensitive data like private keys.

**Comment by sunfishcode** [comment]
The official document for this function is [here](https://github.com/WebAssembly/wasi-filesystem/blob/main/wasi-filesystem.wit.md#change-file-permissions-at). It is admittedly vague, so I filed https://github.com/WebAssembly/wasi-filesystem/issues/74 about documenting it.

When I proposed this function, I was unsure about whether it would make sense to support anything other than global permissions, since WASI has no concept of a "user", however thinking through it now, I think there is a way in which it makes sense. We should document that the actual protection provided may vary widely, including offering no protection at all on some host/filesystems, but nevertheless, the feature may be useful.

I filed https://github.com/WebAssembly/wasi-filesystem/issues/75 to track this.
